### PR TITLE
Bug 1566382 - List error recording values for timespan

### PR DESCRIPTION
**Pull request by badboy**
This also includes documenting the raw API

### CODE REVIEW

File: docs/user/metrics/timespan.md [hunk 1]
```diff
@@ -78,8 +93,10 @@ assertTrue(Auth.loginTime.testGetValue() > 0)
 ## Recorded errors
 
 * `invalid_value`: If recording a negative timespan.
```
Comment (Dexterp37) [inline docs/user/metrics/timespan.md]: nit: could this be a two layers list?

- `invalid_value`:
  - If starting a timer..
  - If stopping a timer...

File: docs/user/metrics/timespan.md [hunk 2]
```diff
@@ -67,6 +67,21 @@ assertTrue(Auth.loginTime.testHasValue())
 assertTrue(Auth.loginTime.testGetValue() > 0)
 ```
 
+### Raw API
+
+> **Note**: It's recommended to use the timer API listed above.
```
Comment (Dexterp37) [inline docs/user/metrics/timespan.md]: nit: maybe we should slightly rephrase this to something like "The raw API was designed to support a specific set of use-cases. Please consider using the higher level APIs listed above."

File: docs/user/metrics/timespan.md [hunk 3]
```diff
@@ -67,6 +67,21 @@ assertTrue(Auth.loginTime.testHasValue())
...
+
+It's possible to explicitly set the timespan value, in nanoseconds.
+This API should only be used if your library or application requires recording times in a way that can not make use of `start`/`stop`/`cancel`.
```
Comment (Dexterp37) [inline docs/user/metrics/timespan.md]: nit: should we add a specific example here? e.g. storing the values of low-level os timers?

Comment (badboy) [reply docs/user/metrics/timespan.md]: Hm, it's currently nowhere used so I have a hard time coming up with an example

Comment (Dexterp37) [reply docs/user/metrics/timespan.md]: Sync was planning on use this to report the time it took for some sync operation to complete on the Rust side. Another example could be the use of some low level API that just exposes elapsed times

File: docs/user/metrics/timespan.md [hunk 4]
```diff
@@ -67,6 +67,23 @@ assertTrue(Auth.loginTime.testHasValue())
...
+> Please consider using the higher level APIs listed above.
+
+It's possible to explicitly set the timespan value, in nanoseconds.
+This API should only be used if your library or application requires recording times in a way that can not make use of `start`/`stop`/`cancel`.
+
+```Kotlin
+import org.mozilla.yourApplication.GleanMetrics.HistorySync
+
+val duration = SyncResult.status.syncs.took.toLong()
+HistorySync.setRawNanos(duration)
```
Comment (badboy) [inline docs/user/metrics/timespan.md]: @Dexterp37 	What do you think about this (kinda constructed) example?

Comment (Dexterp37) [reply docs/user/metrics/timespan.md]: Good enough for me! Let's see what @mdboom thinks about this too ;)
